### PR TITLE
Use FsToolkit.ErrorHandling and reorganise modules

### PR DESCRIPTION
**Pull request by michaelwinch**
These changes update the library to use FsToolkit.ErrorHandling and remove any overlapping code. It also reorganises the modules in the style of FsT by:

- Separating the helper functions, the CE and the operators into separate modules for each permutation of types
- The helper functions now require qualified access
- The CEs are auto opened
- The operators have to be opened explicitly and live in `AsyncWriterResult.Operator.<module>`

What this doesn't include:
- Reorganising the traverse/sequence functions to live in the target modules (i.e. `AsyncWriterResult.traverseResultM` should instead be `List.traverseAsyncWriterResultM`)
- Adding any new helper functions

### CODE REVIEW

**Comment by michaelwinch** [comment]
It could make sense to change the name of the namespace as it could now be a bit confusing, having an `AsyncWriterResult` module inside the `AsyncWriterResult` namespace. If we rename it though it probably makes sense to rename the package which sounds like more effort than it's worth. Thoughts?

Also I ideally want to include this in the next major package version along with some extra functions and changes - what's the best way to go about this? I could create a feature branch on my fork or create a feature branch on the main repo and create my PRs against that?

**Comment by blair55** [comment]
Re the namespace question.. what does FsT do? 😄 

Happy for there to be one big PR if we are aiming for a single new major version with all refactoring and new functionality. But equally fine with merging this as-is to create 1.0.0, followed swiftly by a 1.0.1 with new functions - esp if this is less effort to coordinate re branches etc. Is this even the question you are asking?! 😅

File: src/AsyncWriterResult/AsyncWriterResultOp.fs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+[<AutoOpen>]
+module AsyncWriterResult =
+    /// AsyncWriterResult.map
+    let (<!>) f x = AsyncWriterResult.map f x
+
+    /// AsyncWriterResult.apply
+    let (<*>) f x = AsyncWriterResult.apply f x
+
+    /// AsyncWriterResult.bind
+    let (>>=) x f = AsyncWriterResult.bind f x
```
Comment (blair55) [inline src/AsyncWriterResult/AsyncWriterResultOp.fs:15]: I thought the value of moving the operators into a separate file/module was to give the client the option to `open`. So didn't expect we'd `[<AutoOpen>]` everywhere.

logaf low, but is this done to maintain parity with the existing version and minimise effort after upgrading, or are we following FsT?

Comment (michaelwinch) [reply src/AsyncWriterResult/AsyncWriterResultOp.fs:15]: This is still following FsT - initially I was thinking it had to be this way modules and namespaces work across multiple files but as far as I can tell, there aren't any operators across multiple files.

In this case, we have

```
namespace AsyncWriterResult.Operator.AsyncWriterResult
[<AutoOpen>]
module AsyncWriterResult =
```
which means you can open `AsyncWriterResult.Operator.AsyncWriterResult` which will give you access to the operators, but I don't see how this is different in practice to
```
namespace AsyncWriterResult.Operator
module AsyncWriterResult =
```
🤷

**Comment by michaelwinch** [comment]
> Re the namespace question.. what does FsT do? 😄

FsTs root namespace is `FsToolkit.ErrorHandling` so they get away with not having to worry about it. E.g. one of their functions lives at `FsToolkit.ErrorHandling.Result.map`, but ours live at `AsyncWriterResult.AsyncWriterResult.map` or `AsyncWriterResult.Writer.write`. It's not too big a deal though because we'll just open `FsToolkit.ErrorHandling` or `AsyncWriterResult` anyway.

> Happy for there to be one big PR if we are aiming for a single new major version with all refactoring and new functionality. But equally fine with merging this as-is to create 1.0.0, followed swiftly by a 1.0.1 with new functions - esp if this is less effort to coordinate re branches etc. Is this even the question you are asking?! 😅

Sort of, but you've answered my question anyway! 😄

**Comment by michaelwinch** [comment]
@blair55 I'm gonna go with the option to merge this then follow it up with another PR with new functions; is this okay to merge as it is now? In terms of versioning, the numbering will go from something like 0.0.46 -> 1.0.47

**Comment by michaelwinch** [comment]
Oh of course I'll fix the build failing, not sure why it is though as it's building locally. Will look into it tomorrow

**Review by blair55 — APPROVED** [review]
Sure, all good with me 👍

**Pull request merged**